src/components/timeline/LayerActivity.svelte

```diff
@@ -163,42 +165,45 @@
   }
 
   /**
-   * @note We only allow dragging parent activities. Disable dragging all activities if timeline is "locked"
+   * @note We only allow dragging parent activities.
    */
   function dragActivityStart(points: ActivityPoint[], offsetX: number): void {
-    if (!timelineLocked && points.length) {
+    if (points.length) {
       const [point] = points;
       if (point.parent_id === null) {
         dragOffsetX = offsetX - xScaleView(point.x);
         dragPoint = point;
+        dragStartX = dragCurrentX = dragPoint.x;
       }
     }
   }
 
   function dragActivity(offsetX: number): void {
-    if (dragOffsetX && dragPoint) {
+    // Only update the x position if the timeline is unlocked
+    if (!timelineLocked && dragPoint) {
       const x = offsetX - dragOffsetX;
-      const unixEpochTime = xScaleView.invert(x).getTime();
-      const start_time_doy = getDoyTime(new Date(unixEpochTime));
-      if (unixEpochTime !== dragPoint.x) {
+      dragCurrentX = xScaleView.invert(x).getTime();
+      const start_time_doy = getDoyTime(new Date(dragCurrentX));
+      if (dragCurrentX !== dragPoint.x) {
         $activitiesMap[dragPoint.uniqueId].start_time_doy = start_time_doy;
```

Comment:
I actually managed to break this line (though it's not part of your PR) by holding shift, dragging, and also then pressing delete key and then dragging again. It's possible that the activity no longer exists during a drag. But a very weird edge case.

Reply:
But the result is an unresponsive canvas so might be worth guarding against?

Reply:
ah interesting! I guess at any drag update/end event we should just double check the activity still exists?

Reply:
Yeah seems like the safest thing to do